### PR TITLE
Add tests for offline functionality

### PR DESCRIPTION
**Pull request by Saela**
- Adds some tests to check if showing/hiding of elements works for offline/online switch
- Local DB Service couldn't be tested as I could not correctly mock the dexie config 👎

### CODE REVIEW

File: __tests__/unit/components/app/TourForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { render } from '@testing-library/react'
+import { MemoryRouter } from 'react-router'
+
+const mockConnectionStatusContext: any = {
+  isOffline: () => false
+}
+const mockImageUploadContext: any = {
+  files: [],
+  currentUploads: []
+}
```
Comment (Tugark) [inline __tests__/unit/components/app/TourForm.test.tsx:17]: Why not typehint the object properly? You could still only specify those props you need by using `as XyzContext`.

File: __tests__/unit/components/providers/ConnectionStatusProvider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    const { queryByTestId, getByTestId } = renderTestComponent()
+    fireEvent.click(getByTestId('update-connection-status-button--online'))
+    const testElement = queryByTestId('isOffline')
+    expect(testElement!.innerHTML).toEqual('false')
+    const storedConnectionStatus = localStorage.getItem(LocalStorageKey.ConnectionStatus)
+    expect(storedConnectionStatus).toEqual(ConnectionStatus.ONLINE)
+    const goOnlineButton = queryByTestId('showButton')
+    expect(goOnlineButton!.innerHTML).toEqual('false')
+    const storedButtonVisibility = localStorage.getItem(LocalStorageKey.IsGoOnlineButtonVisible)
+    expect(storedButtonVisibility).toEqual('false')
```
Comment (Tugark) [inline __tests__/unit/components/providers/ConnectionStatusProvider.test.tsx:77]: I don't mind this, but our other tests all have `expect` at the end (kind of Arrange-Act-Assert).

File: __tests__/unit/components/shared/navbar/GoOnlineButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+const triggerSyncFailedNotification:jest.Mock = jest.fn()
+const mockNotificationContext: any = {
+  triggerSyncFailedNotification
+}
+
+const mockAuthContext: any = {
+  token: '1234'
+}
```
Comment (Tugark) [inline __tests__/unit/components/shared/navbar/GoOnlineButton.test.tsx:21]: See comment above.

File: jest.config.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
+const { jsWithBabelESM: tsjPreset } = require('ts-jest/presets')
 const customJestConfig = {
   preset: 'ts-jest',
   moduleDirectories: ['node_modules', '<rootDir>/'],
   testEnvironment: 'jest-environment-jsdom',
   transform: {
+    ...tsjPreset.transform,
```
Comment (Tugark) [inline jest.config.js:7]: What kind of black-magicery is this? :D

**Review by Tugark — COMMENTED** [review]
LGTM

File: __tests__/unit/components/app/TourForm.test.tsx (hunk 1 above)
Comment (Saela) [reply __tests__/unit/components/app/TourForm.test.tsx:17]: How would you do that so I can only specify the props I need? Adding the specific type will always ask for all of the properties to be defined unless you use some hacky solution. Or at least that's what I experienced so far. Let's talk about this in one of our meetings. :)

File: jest.config.js (hunk 4 above)
Comment (Saela) [reply jest.config.js:7]: It's from the ts-jest documentation: https://kulshekhar.github.io/ts-jest/docs/getting-started/presets/

File: __tests__/unit/components/providers/ConnectionStatusProvider.test.tsx (hunk 2 above)
Comment (Saela) [reply __tests__/unit/components/providers/ConnectionStatusProvider.test.tsx:77]: Fixed in: https://github.com/gipfeli-io/gipfeli-frontend/pull/120

**Pull request merged**